src/cli.rs

```diff
@@ -585,13 +587,33 @@ async fn run_deploy(
         )
     };
 
-    for data in data_iter() {
-        deploy::push::build_profile(data).await?;
-    }
+    let (remote_builds, local_builds): (Vec<_>, Vec<_>) = data_iter().partition(|data| {
+        data.deploy_data.merged_settings.remote_build.unwrap_or_default()
+    });
+
+    // await both the remote builds and the local builds to speed up deployment times
+    try_join!(
+        // remote builds can be run asynchronously since they do not affect the local machine
+        try_join_all(remote_builds.into_iter().map(|data| async {
```

Comment:
From `try_join_all` docs:
> If any future returns an error then all other futures will be canceled and an error will be returned immediately

IMO, it might be better to wait for all futures to be completed/failed instead of canceling everything on the first failure, thus the non-failed builds will complete despite the error in an unrelated build. What do you think?

Reply:
> Since the builds are started concurrently, the build progress information is pretty much useless and flickers a lot.

I'm not sure how viable it is to implement, but perhaps, it's possible to collect `stdout` and `stderr` for each future, if it is, we can collect and report detailed output synchronously once all remote builds are completed

Reply:
> IMO, it might be better to wait for all futures to be completed/failed instead of canceling everything on the first failure, thus the non-failed builds will complete despite the error in an unrelated build. What do you think?

I'll see how easy it would be to implement that, if it is that might be worth checking out so that consecutive builds don't have to build as much.

> perhaps, it's possible to collect stdout and stderr for each future, if it is, we can collect and report detailed output synchronously once all remote builds are completed

I was considering that maybe one line from the bottom per build would be neat e.g.:

[previous output]
host 1: [build progress] 
host 2: [build progress]

This will, however, require some more fine grained control over the terminal output i.e. raw output instead of cooked.

Reply:
> would be neat

Agree, but indeed sounds quite non-trivial

Reply:
Oh, another somewhat conceptual concern. Profiles can potentially depend on each other, so perhaps it's worth doing the parallelization on per-host basis instead of pre-profile